### PR TITLE
(Feature) Ceramic feed dashboard: Implement new publish flow, merge with backend in a single app

### PR DESCRIPTION
**Pull request by johnsmith-gooddollar**
# Description

- merged with ceramic backend
- added ceramic client
- added ceramic models
- updated post to use draft feature
- removed rest client (direct ceramic connection only)
- added cli scripts to generate DID and export React app settings
- afterUpdate handler

About #3666

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/plugins/ceramic-feed/server/content-types/ceramic-post/lifecycles.js [hunk 1]
```diff
@@ -15,34 +17,11 @@ const LifecycleHooks = new class {
...
-      payload[field] = filePath
-    }))
-
-    const { id } = await ceramic.create(payload)
-
-    data.cid = id
+    assign(this, { strapi, fields, mediaFields })
   }
 
   async afterUpdate({ params, result }) {
```
Comment (sirpy) [inline src/plugins/ceramic-feed/server/content-types/ceramic-post/lifecycles.js]: the logic is:
is the publishedAt date set, then update ceramic
if publishedAt date is null then you need to read old post and check if it was previously published and now it is being unpublished. If it wasnt previously published then do nothing... do not update ceramic. if was previously published need to mark as deleted/unpublished on ceramic

Comment (sirpy) [reply src/plugins/ceramic-feed/server/content-types/ceramic-post/lifecycles.js]: Also I dont see a match for `ceramic.update(data.cid,payload)` what is CID? where is the ceramic api receiving a cid?

**Review by sirpy — CHANGES_REQUESTED** [review]
when we went over strapi i've also shown you that I use a version that works with typescript

**Comment by johnsmith-gooddollar** [comment]
1. Had to do it asap, so the default template was used. Will add task for TS migration and assign it to @alt-a1r 
I do not prefer use TS when I'm limited in time because it have a huge overhead due to the types matching. Sometimes it's more strict than Java and I have feeling I'm trying to code in C++ ;) 
2. If post **is published **and updates - we're updating it in ceramic.
3. If post **is draft** and updated in Strapi - we're ignoring this
4. If publishedAt is present in the data event prop that means publish or unpublish button was pressed. No other options possible. If publishedAt is null it's means we have to unpublish definitely. There no way to unpublish something twice (there's no unpublish button for draft - only publish / update and vice versa). 
5. Also there're additional checks on ceramic side. We won't try to republish or unpublish post twice. Also the same document will be used if unpublish post then re-publish it again (because the unique once-generated cid is kept in the Strapi db)
6. Tomorrow I'll append this with the comments
7. You reviewed the old code without the latest commit where all the logic we discussed was implemented

**Comment by johnsmith-gooddollar** [comment]
@sirpy i've created task for TS migration: https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/3682
We need to discuss it and clarify which TS template to use because there's no official typescript bolierplate yet

**Pull request merged**